### PR TITLE
support multiple layer yaml in fragments

### PR DESCRIPTION
**Pull request by 928PJY**
support multiple layer yaml in fragments

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.OverwriteDocuments/OverwriteDocumentModelCreater.cs [hunk 1]
```diff
@@ -167,32 +152,40 @@ private void FindOrCreateObject(Dictionary<string, object> currentObject, Block
...
-                    // Duplicate
+                    // Duplicate OPath in markdown section
                     Logger.LogWarning(
                         $"There are two duplicate OPaths {originalOPathString}, the previous one will be overwritten",
                         line: codeHeaderBlock.Line.ToString(),
-                        code: WarningCodes.Overwrite.InvalidOPaths);
+                        code: WarningCodes.Overwrite.InvalidMarkdownFragments);
                 }
-                else
+                else if(value is Dictionary<object, object> || value is List<object>)
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.OverwriteDocuments/OverwriteDocumentModelCreater.cs]: >if [](start = 21, length = 2)

space after if

File: src/Microsoft.DocAsCode.Build.OverwriteDocuments/OverwriteDocumentModelCreater.cs [hunk 2]
```diff
@@ -167,32 +152,40 @@ private void FindOrCreateObject(Dictionary<string, object> currentObject, Block
...
                 {
                     throw new MarkdownFragmentsException(
                         $"A({lastSegment.SegmentName}) is expected to be an dictionary like \"A/B\" or an array of dictionaries like \"A[c=d]/C\", however it is used as an array of Blocks in line {codeHeaderBlock.Line} like \"../A\" OPath syntax",
                         codeHeaderBlock.Line);
                 }
+                else
+                {
+                    // Duplicate OPath in yaml section and markdown section
+                    Logger.LogWarning(
+                        $"There are two duplicate OPaths `{originalOPathString}` in yaml code block and contents block, and markdown property is not allowed inside a YAML code block",
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.OverwriteDocuments/OverwriteDocumentModelCreater.cs]: >, and markdown property is not allowed inside a YAML code block [](start = 118, length = 63)

remove

Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.OverwriteDocuments/OverwriteDocumentModelCreater.cs]: >There are [](start = 26, length = 9)

remove

File: test/Microsoft.DocAsCode.Build.OverwriteDocuments.Tests/OverwriteDocumentModelCreaterTest.cs [hunk 3]
```diff
@@ -24,17 +26,20 @@ public class OverwriteDocumentModelCreaterTest
...
-            Assert.Equal("f", actual["e"]);
+            var yamlCodeBlockString = @"name: name overwrite
+definitions:
+- name: Application 1
+  properties:
+  - name: id
+    description: overwrite in yaml block
+  - name: displayName
+    description: overwrite in yaml block";
+            var testYamlCodeBlock = Markdown.Parse($"```\n{yamlCodeBlockString}\n```")[0];
```
Comment (vwxyzh) [inline test/Microsoft.DocAsCode.Build.OverwriteDocuments.Tests/OverwriteDocumentModelCreaterTest.cs]: ```` ```yaml ````

**Pull request merged**